oaknational/oak-ai-lesson-assistant
# chore: delete old google drive export files cron job- AI-561

## Description

- Creates an expired exports route for a vercel [cron job](https://vercel.com/docs/cron-jobs/quickstart)
The route will fetch google drive ids that are expired (over two weeks) and update the expiredAt field in lesson_exports where the gdriveFileIds match. It then deletes the files from drive.
- This cron job can be scheduled to run overnight (settings in vercel.json) - Not sure how regularly we want to do this currently set every day at 3am
```
"schedule": "0 3 * * *" 
minute hour day month day-of-week 
```


## How to test

Create a lesson and  export resources.
Change the query so that it fetches all files not just older than 14 days

```
    // const query = `modifiedTime < '${oneMonthAgo}' and '${folderId}' in parents`;
    const query = `'${folderId}' in parents`;

```

Run app locally and run the cron job by going to http://localhost:2525/api/cron-jobs/expired-exports

Check that files are deleted from Google Drive playground output folder and `expired_at` has been set to current time in local database.

## Checklist
